test/1.4/lib/T_signal_templates.py

```diff
@@ -0,0 +1,84 @@
+# © 2022 Intel Corporation
+# SPDX-License-Identifier: MPL-2.0
+
+import os
+import simics
+import stest
+import tempfile
+
+
+class signal_stub:
+    cls = simics.confclass('signal-stub')
+    cls.attr.level('i', default=0)
+
+    @cls.iface.signal.signal_raise
+    def signal_raise(self):
+        self.level += 1
+
+    @cls.iface.signal.signal_lower
+    def signal_lower(self):
+        self.level -= 1
+
+
+# signal level high at init
+stub = simics.pre_conf_object('stub', 'signal-stub')
+obj = simics.pre_conf_object('test', 'test',
+                             outsig=stub,
+                             outsig_signal_high=True)
+simics.SIM_add_configuration([stub, obj], None)
+stub = simics.SIM_get_object(stub.name)
+obj = simics.SIM_get_object(obj.name)
+stest.expect_equal(stub.level, 1)
+
+# loading a checkpoint doesn't call raise
+with tempfile.TemporaryDirectory() as d:
```

Comment:
IIRC, the TemporaryDirectory will be deleted when you exit the block; better for debug to create a dir under scratchdir that survives the test run.

Reply:
stest.scratch didn't seem reliable when I did `make test-dmlc`, it tried to create the checkpoint in `./scratch/checkpoint` which didn't exist.

Wherever stest.scratch decides to put files is a mystery to me, finding it after a failed test would take significantly longer for me than to rerun the test locally after modifying the test to use a hardcoded path for the checkpoint.

Reply:
Grepping, it seems that there's a global variable `scratchdir` that you can access from `T_*.py`. Some other tests do that.

Reply:
Sure, if that's what you want. I still won't know where `scratchdir` is so it makes no huge difference to me :-)